File: Arrays and Hashing/longest-consecutive-sequence.py
# time O(n) space O(n)
class Solution:
    def longestConsecutive(self, nums):
        hash_set = set(nums)
        longest = 0
        for num in nums:
            if num - 1 not in hash_set:
                length = 0
                while(length + num) in hash_set:
                    length += 1
                if length > longest:
                    longest = length
        return longest

File: Arrays and Hashing/longest-consecutive-sequence.js
// time O(n) space O(n)
var longestConsecutive = function (nums) {
  let hashSet = new Set(nums);
  let longest = 0;
  for (let num of nums) {
    let length = 0;
    if (!hashSet.has(num - 1)) {
      while (hashSet.has(length + num)) {
        length++;
      }
    }
    if (length > longest) {
      longest = length;
    }
  }
  return longest;
};
